examples/tutorials/tutor.py

```diff
@@ -12,1692 +12,45 @@
-                show_labels=False,
-            ),
-        ),
-        title="Python Tutor",
-        id="dmorrill.tutor.tutor:1.0",
-        buttons=NoButtons,
-        resizable=True,
-        width=0.8,
-        height=0.8,
-    )
-
-    def _path_changed(self, path):
-        """ Handles the tutorial root path being changed.
-        """
-        self.init_tutor()
-
-    def _next_changed(self):
-        """ Displays the next tutorial section.
-        """
-        self.section = self.next_section
-
-    def _previous_changed(self):
-        """ Displays the previous tutorial section.
-        """
-        self.section = self.previous_section
-
-    def _parent_changed(self):
-        """ Displays the parent of the current tutorial section.
-        """
-        self.section = self.section.parent
-
-    def _reload_changed(self):
-        """ Reloads the tutor from the original path specified.
-        """
-        self.init_tutor()
-
-    @cached_property
-    def _get_next_section(self):
-        """ Returns the next section of the tutorial.
-        """
-        next = None
-        section = self.section
-        if len(section.subsections) > 0:
-            next = section.subsections[0]
-        else:
-            parent = section.parent
-            while parent is not None:
-                index = parent.subsections.index(section)
-                if index < (len(parent.subsections) - 1):
-                    next = parent.subsections[index + 1]
-                    break
-
-                parent, section = parent.parent, parent
-
-        return next
-
-    @cached_property
-    def _get_previous_section(self):
-        """ Returns the previous section of the tutorial.
-        """
-        previous = None
-        section = self.section
-        parent = section.parent
-        if parent is not None:
-            index = parent.subsections.index(section)
-            if index > 0:
-                previous = parent.subsections[index - 1]
-                while len(previous.subsections) > 0:
-                    previous = previous.subsections[-1]
-            else:
-                previous = parent
-
-        return previous
-
-    def _get_title(self):
-        """ Returns the title of the current section.
-        """
-        section = self.section
-        if section is None:
-            return ""
-
-        return "%s: %s" % (section.__class__.__name__, section.title)
-
-    def init_tutor(self):
-        """ Initials the tutor by creating the root section from the specified
-            path.
-        """
-        path = self.path
-        title = title_for(os.path.splitext(os.path.basename(path))[0])
-        section = SectionFactory(title=title).trait_set(path=path).section
-        if section is not None:
-            self.section = self.root = section
 
+def main(root_dir):
+    # Create a tutor and display the tutorial:
+    path, name = os.path.splitext(root_dir)
+    demo(dir_name=root_dir, title='Traits Demos')
```

Comment:
A couple of things I noticed while testing the runner:
1. Description of `doc_examples` folder shows `<img src="traits_ui_demo.jpg">`, which is programmed to be default description if there’s no `__init__.py` file.
2. Description of  `examples` folder shows a thrown error:
```
FileNotFoundError: [Errno 2] No such file or directory: 'doc_examples/examples/__init__.py'
```
Do we want to add these `__init__.py` files with descriptions, should this be addressed in `traitsui` code or is it okay to be like this for now?

Reply:
Include the `__init__.py` and add a brief explanation of what is in the directory.

Reply:
Added both `__init__.py` files but the one in`doc_examples` isn't shown right now. To show it, `use_files=True` is needed, but that also picks up `doc_examples.rst` and with the new tree structure https://github.com/enthought/traitsui/issues/792 bug is exposed. So I think it's better to show a reference to `traits_ui_demo.jpg` until the bug is fixed.

The explanations I added are very minimal. Didn't want to add the actual contents because that would require manual upkeeping, so I hope this is enough.

Reply:
@corranwebster Why do we want an `__init__.py` file in `doc_examples`? That directory is not a Python package. Is there some part of the machinery in `tutor.py` that needs this?

Reply:
`tutor.py` machinery looks for `__init__.py` of directories in order to extract their descriptions. It is optional for the root directory, with default description set to `<img src="traits_ui_demo.jpg">`, but all other subdirectories are expected to have `__init__.py` or the code raises a non-fatal error.

Reply:
Got it. Thanks.

That's not nice, but it is what it is. There's definitely scope for rewriting and improving `tutor.py` one day. But not today.